codeworld-api/src/CodeWorld/Picture.hs

```diff
@@ -57,6 +57,15 @@ data Picture
     | ThickPolygon CallStack
            [Point]
            !Double
+    | Rectangle CallStack
```

Comment:
Let's start with getting these constructors right.  The arguments to each constructor in `Picture` should be a `CallStack`, and then the same arguments as the API function.  So `Rectangle` and `SolidRectangle` shouldn't have a list of points, just a width and height.  `ThickRectangle` needs a width, height, and line thickness.  Once you have that information, that's already enough to find the points!

I think you had this right at first, but instead of pushing forward and fixing the rest of the code, you keep going back and changing what you've already done.  That makes it hard to figure out where you're confused, and what you know.  So let's start here; get these types right, and the rest should follow from that.

Reply:
I think I'm a lot closer now, but just one more error to fix regarding points not being in scope in Driver.hs (lines 489, 500 and 505) (from last commit)

Reply:
I made a second version that has "width" and "linewidth".